sdk/core/Azure.Core/samples/Serialization.md

```diff
@@ -0,0 +1,164 @@
+# Azure.Core public serialization samples
```

Comment:
If this is coming in a separate PR, would it make sense to remove this from this PR?  Merging this as part of this PR will make it public documentation, and I am not sure this version is what we want to release initially.